lib/std/heap.zig

```diff
@@ -886,6 +886,21 @@ fn testAllocator(allocator: *mem.Allocator) !void {
         allocator.destroy(item);
     }
 
+    slice = allocator.shrink(slice, 200);
+    testing.expect(slice.len == 200);
+
+    for (slice) |*item, i| {
+        item.* = try allocator.dupeOne(@intCast(i32, i * 50));
+    }
+
+    slice = try allocator.realloc(slice, 15000);
+    testing.expect(slice.len == 15000);
+
+    for (slice[0..200]) |item, i| {
+        testing.expect(item.* == @intCast(i32, i * 50));
+        allocator.destroy(item);
+    }
+
```

Comment:
More tests with complex types?

Reply:
Good idea, but I just added to the existing allocator testing infrastructure.

Ping me in IRC to remind me later :)